torch/csrc/jit/codegen/cuda/root_domain_map.cpp

```diff
@@ -297,6 +297,10 @@ void UnmappableReductionDomains::handleReductionOutput(TensorView* out_tv) {
   auto use_chains = DependencyCheck::getAllUseChains(out_tv);
   for (const auto& chain : use_chains) {
     for (const auto& tv : ir_utils::filterByType<TensorView>(chain)) {
+      // Do not include the tensor itself in its consumers
+      if (tv == out_tv) {
+        continue;
+      }
```

Comment:
This is another related fix. Here, we are gathering IterDomains of all consumer tensors of a tensor with a reduction IterDomain. For this logic, we don't want to have the same tensor as its consumer.